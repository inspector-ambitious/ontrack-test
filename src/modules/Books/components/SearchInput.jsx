import React, { useState } from "react";
import PropTypes from "prop-types";
import { Row, Col, Button, Form, InputGroup } from "react-bootstrap";

const SearchInput = ({ setSearchQuery, searchQuery, setPage }) => {
  const [currentValue, setCurrentValue] = useState(searchQuery);

  return (
    <Row className="mb-4 mt-4">
      <Col lg={4}>
        <Form>
          <InputGroup>
            <Form.Control
              type="search"
              placeholder="Search..."
              onChange={(event) => {
                setCurrentValue(event.target.value);
              }}
              onKeyPress={(event) => {
                if (event.key === "Enter") {
                  setSearchQuery(currentValue);
                  setPage(1);
                }
              }}
            />
            <Button
              className="ml-2"
              onClick={() => {
                setSearchQuery(currentValue);
                setPage(1);
              }}
            >
              Search
            </Button>
          </InputGroup>
        </Form>
      </Col>
    </Row>
  );
};

SearchInput.propTypes = {
  setSearchQuery: PropTypes.func.isRequired,
  setPage: PropTypes.func.isRequired,
  searchQuery: PropTypes.string,
};

export default SearchInput;
