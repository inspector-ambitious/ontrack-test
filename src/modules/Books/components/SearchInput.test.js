import React from "react";
import { shallow } from "enzyme";
import SearchInput from "./SearchInput";

describe("Books/components/SearchInput", () => {
  const setPage = jest.fn();
  const setSearchQuery = jest.fn();
  const defaultProps = {
    setPage,
    searchQuery: "foobar",
    setSearchQuery,
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("should render without crashing", () => {
    const wrapper = shallow(<SearchInput {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  describe("when search button is clicked", () => {
    it("should call setSearchQuery", () => {
      const wrapper = shallow(<SearchInput {...defaultProps} />);
      wrapper.find("Button").simulate("click");
      expect(setSearchQuery).toHaveBeenCalledTimes(1);
      expect(setSearchQuery).toHaveBeenCalledWith("foobar");
    });
    it("should set the page to 1", () => {
      const wrapper = shallow(<SearchInput {...defaultProps} />);
      wrapper.find("Button").simulate("click");
      expect(setSearchQuery).toHaveBeenCalledTimes(1);
      expect(setPage).toHaveBeenCalledWith(1);
    });
  });
  describe("when enter is pressed", () => {
    it("should call setSearchQuery", () => {
      const wrapper = shallow(<SearchInput {...defaultProps} />);
      wrapper.find("FormControl").simulate("keypress", { key: "Enter" });
      expect(setSearchQuery).toHaveBeenCalledTimes(1);
      expect(setSearchQuery).toHaveBeenCalledWith("foobar");
    });
    it("should set the page to 1", () => {
      const wrapper = shallow(<SearchInput {...defaultProps} />);
      wrapper.find("FormControl").simulate("keypress", { key: "Enter" });
      expect(setSearchQuery).toHaveBeenCalledTimes(1);
      expect(setPage).toHaveBeenCalledWith(1);
    });
  });
  describe("when the user change the value and press enter", () => {
    it("should call setSearchQuery with the appropriate value", () => {
      const wrapper = shallow(<SearchInput {...defaultProps} />);
      wrapper
        .find("FormControl")
        .simulate("change", { target: { value: "baz" } });
      wrapper.find("FormControl").simulate("keypress", { key: "Enter" });
      expect(setSearchQuery).toHaveBeenCalledTimes(1);
      expect(setSearchQuery).toHaveBeenCalledWith("baz");
    });
  });
});
