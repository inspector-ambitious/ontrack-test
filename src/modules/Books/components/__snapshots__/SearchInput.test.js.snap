// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books/components/SearchInput should render without crashing 1`] = `
<Row
  className="mb-4 mt-4"
  noGutters={false}
>
  <Col
    lg={4}
  >
    <Form
      inline={false}
    >
      <ForwardRef(InputGroup)>
        <FormControl
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Search..."
          type="search"
        />
        <Button
          active={false}
          className="ml-2"
          disabled={false}
          onClick={[Function]}
          variant="primary"
        >
          Search
        </Button>
      </ForwardRef(InputGroup)>
    </Form>
  </Col>
</Row>
`;
